cpp/oneapi/dal/algo/kmeans_init/common.hpp

```diff
@@ -74,6 +74,12 @@ constexpr bool is_valid_method_v = dal::detail::is_one_of_v<Method,
 template <typename Task>
 constexpr bool is_valid_task_v = dal::detail::is_one_of_v<Task, task::init>;
 
+template <typename M>
+constexpr bool is_not_default_dense = !std::is_same_v<M, method::dense>;
+
+template <typename M>
+using enable_if_not_default_dense = std::enable_if_t<is_not_default_dense<M>>;
+
```

Comment:
Those redefinitions look like an overkill for me. They do not make the code simpler, but they require additional effort from reader to understand what is hidden underneath "enable_if_not_default_dense".

Reply:
I believe it contains everything you need t know just in naming.

Reply:
We do not need to agree on how particularly each line of code is implemented. We can have different sense of beauty, it is Ok.
In general the code looks good to me, I approve the changes.